File: gestao-spring/src/main/java/com/erp/gestao/dto/filterDTO/ReservaFilterDTO.java
package com.erp.gestao.dto.filterDTO;

import com.erp.gestao.model.Produto;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class ReservaFilterDTO {
        private Integer id;
        private Produto produto;
        private LocalDate dataReserva;
        private Integer quantidade;
        private Boolean estorno = false;
        private Integer quantidadeEstornada;
        private LocalDate dataEstorno;
        private Boolean estornoCompleto = false;
        private Integer versao = 0;
}

File: gestao-angular/src/app/shared/dto/filterDTO/reserva.filter.dto.ts
import {Produto} from "../../model/produto";
import {Pessoa} from "../../model/pessoa";

export class ReservaFilterDto {
  id: number;
  locatario: Pessoa;
  produto: Produto;
  dataReserva: Date;
  quantidadeDeDias: number;
  previsaoDeEntrega: Date;
  quantidadeLocada: number;
  estorno: boolean = false;
  quantidadeEstornada: number;
  dataEstorno: Date;
  estornoCompleto: boolean = false;
  versao: number;
}

